app/public/js/localSprites.js

```diff
@@ -308,33 +304,46 @@ function addTriggerZone(left, top, width, height, target, spawnedObjectType, spa
         //$.extend(rectangle(3 * 8, 0 * 8, 8, 8), {image: fireballContrailAImage, hitBox}),
         //$.extend(rectangle(4 * 8, 0 * 8, 8, 8), {image: fireballContrailAImage, hitBox}),
     ];
-    var triggerZone = new SimpleSprite({frames}, left, top, 0, 0, 1, 1);
-    triggerZone.name = 'triggerZone',
-    triggerZone.hitBox = rectangle(left, top, width, height),
-    triggerZone.target = target,
-    triggerZone.cooldownFrames = cooldownFrames,   //50 per second, I think
-    triggerZone.cooldownTimer = 0,
-    triggerZone.xForce = xForce,
-    triggerZone.yForce = yForce,
-    triggerZone.spawnedObjectType = 1,
-    triggerZone.spawnedObjectXOffset = (left + width / 2) + spawnedObjectXOffset,
-    triggerZone.spawnedObjectYOffset = (top + width / 2) + spawnedObjectYOffset,
-    triggerZone.framesToLive = 32767;   //would like to have an easy way to make this infinite, like a -1.
-    localSprites.push(triggerZone);
+    var trigger = new SimpleSprite({frames}, left, top, 0, 0, 1, 1);
+    trigger.type = 'SPRITE_TYPE_TRIGGER',
+    trigger.hitBox = rectangle(left, top, width, height),
+    trigger.target = target,
+    trigger.cooldownFrames = cooldownFrames,   //50 per second, I think
+    trigger.cooldownTimer = 0,
+    trigger.xForce = xForce,
+    trigger.yForce = yForce,
+    trigger.spawnedObjectType = spawnedObjectType,
+    trigger.spawnedObjectXOffset = (left + width / 2) + spawnedObjectXOffset,
+    trigger.spawnedObjectYOffset = (top + width / 2) + spawnedObjectYOffset,
+    trigger.framesToLive = 32767;   //temporary: should change the shouldBeRemoved code to have a more sensible way to signify immortality.
+    localSprites.push(trigger); //this will gone when tigger zones are broken up into types better
 }
 
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
+function addPowerup(x, y, powerupType, xScale, yScale, target, falls) {
```

Comment:
So here you would pass in the actual const value when you call this method, for example:
`addPowerup(x, y, POWERUP_TYPE_HEART, 1, 1, mainCharacter, true)` 
and the below you could set `powerup.powerupType = powerupType;` and then check:
```
if (powerup.powerupType === POWERUP_TYPE_HEART) {
// Do heart powerup specific logic
}
```
That way you don't have to keep track that the heart is value `0` etc.